src/function/scalar/string/bar.cpp

```diff
@@ -0,0 +1,100 @@
+#include "duckdb/common/exception.hpp"
+#include "duckdb/common/operator/cast_operators.hpp"
+#include "duckdb/common/types/string_type.hpp"
+#include "duckdb/common/types/value.hpp"
+#include "duckdb/common/types/vector.hpp"
+#include "duckdb/common/vector_operations/quaternary_executor.hpp"
+#include "duckdb/common/vector_operations/ternary_executor.hpp"
+#include "duckdb/function/scalar/string_functions.hpp"
+
+namespace duckdb {
+
+static string_t BarScalarFunction(double x, double min, double max, double max_width, vector<char> &result) {
+	// U+2588 "█" full
+	constexpr static char FULL_BAR[] = "█";
+	// U+258F "▏" one eighth
+	// U+258E "▎" one quarter
+	// U+258D "▍" three eighths
+	// U+258C "▌" half
+	// U+258B "▋" five eighths
+	// U+258A "▊" three quarters
+	// U+2589 "▉" seven eighths
+	// 7 elements: 1/8, 2/8, 3/8, 4/8, 5/8, 6/8, 7/8
+	constexpr static char FRACTIONAL_BARS[] = "▏▎▍▌▋▊▉";
+	constexpr static idx_t GRADES_IN_FULL_BAR = 8;
+	static const idx_t UNICODE_BAR_CHAR_SIZE = strlen("█");
+
+	if (!Value::IsFinite(max_width)) {
```

Comment:
Using `double` is not meaningless because `max_width` can be fractional like 1 + 1/8, 1 + 1/4, 1 + 3/8, ...
Though I'm not sure whether someone wants to do that.